File: POO_2026_1/src/Abr_23_Menu_Principal/OpcoesMenu.java
package Abr_23_Menu_Principal;

import MeuPacote.Aluno;
import MeuPacote.Geral;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class OpcoesMenu {
  
  public static void exportar(ArrayList<String> alunos) {
    try {
      FileWriter arq = new FileWriter("alunos.txt");
      PrintWriter gravarArq = new PrintWriter(arq);
    
      for(String item: alunos) {
        gravarArq.printf("%s\n", item);
      }
    
      gravarArq.close();
    }
    catch(IOException e) {
      System.out.println("Erro: Exportação dos dados!!!");
    }
  }
  
  public static void importar(ArrayList<String> alunos) {
    try {
      FileReader arq = new FileReader("alunos.txt");
      BufferedReader lerArq = new BufferedReader(arq);
    
      String linha;
    
      linha = lerArq.readLine(); // lê a primeira linha do arquivo
      while (linha != null) {
        alunos.add(linha);
      
        linha = lerArq.readLine(); // lê da segunda até a última linha do arquivo
      }
    
      lerArq.close();
    }
    catch (IOException e) {
      System.out.println("Arquivo 'alunos.txt' está vazio!!!");
    }
  }
  
  public static void cadastrar(ArrayList<String> alunos) {
    Scanner ler = new Scanner(System.in);
    
    String nome;
    double freq, nf;
    
    while (true) {
      System.out.println("Nome (FIM para encerrar):");
      nome = ler.nextLine();
      if (nome.equalsIgnoreCase("FIM"))
          break;
      
      System.out.println();
      System.out.print("Frequência: ");
      freq = ler.nextDouble();
      
      System.out.print("Nota Final: ");
      nf = ler.nextDouble();
      
      System.out.println();
      
      if (Geral.confirmou("Confirma os dados (S/*)? ")) {
         alunos.add(nome + ";" + freq + ";" + nf);
         System.out.println("<<< Cadastrado >>>");
      }
      else {
         System.out.println("<<< \"Não\" cadastrado >>>");
      }
      
      ler.nextLine(); // esvazia o buffer de entrada de dados
      
      System.out.println();
    }

    System.out.println();
  }
  
  public static void listar(ArrayList<String> alunos) {
    int i, n = alunos.size();
    String nome, partes[];
    double freq, nf;
    for(i=0; i<n; i++) {
      partes = alunos.get(i).split(";");
      
      nome = partes[0];
      freq = Double.parseDouble(partes[1]);
      nf = Double.parseDouble(partes[2]);
      
      System.out.printf("%do. aluno\n", (i+1));
      System.out.println(Aluno.toString(nome, nf, freq)); 
      System.out.println();
    }
    
    System.out.println();
  }
  
  public static void excluir(ArrayList<String> alunos) {
    Scanner ler = new Scanner(System.in);
    int i, n = alunos.size();
    
    while (true) {
      listar(alunos);
      
      do {
        System.out.print("Ordem (-1 para encerrar): ");
        i = ler.nextInt();
      } while ((i < -1) || (i > n) || (n == 0));
      
      if (i == -1)
          break;      
      
      System.out.println(alunos.get(i-1));
      if (Geral.confirmou("Excluir (S/*)? ")) {
         alunos.remove(i-1);
         System.out.println("<<< Excluído >>>");
      }
      else {
         System.out.println("<<< Mantido >>>");  
      }
      System.out.println();
    }
  }
  
}

File: CPP/Abr_23_MenuPrincipal_CPP/OpcoesMenu.cpp
#include <string>
#include "OpcoesMenu.h"
#include "Geral.h"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\Abr_14_Usando_MeuPacote_CPP\Usando_Aluno\Aluno.cpp"

using namespace std;

void exportar(vector<string> alunos) {
  FILE *arq;
  arq = fopen("alunos.txt", "w");
  if (arq == NULL) {
     cout << "Erro: Exportação dos dados!!!" << endl << endl;
  }
  else {
    for(string item: alunos) {
      fprintf(arq, "%s\n", item.c_str());
    }
  }

  fclose(arq);
}

void importar(vector<string> *alunos) {
  FILE *arq;
  arq = fopen("alunos.txt", "r");
  if (arq == NULL) {
     cout << "Arquivo 'alunos.txt' está vazio!!!" << endl << endl;
  }
  else {
    char item[100];
    while (fscanf(arq, "%[^\n]s", item) != EOF) {
      fgetc(arq);
      alunos->push_back(item);
    }

    fclose(arq);
  }
}

void cadastrar(vector<string> *alunos) {
  cls();

  string nome;
  double freq, nf;

  while (true) {
    esvaziar();

    cout << "Nome (FIM para encerrar):" << endl;
    getline(cin, nome);
    if (nome == "FIM")
       break;

    cout << endl;
    cout << "Frequência: ";
    cin >> freq;

    cout << "Nota Final: ";
    cin >> nf;

    cout << endl;

    if (confirmou("Confirma os dados (S/*)? ")) {
       alunos->push_back(nome.c_str());
       alunos->push_back(to_string(freq));
       alunos->push_back(to_string(nf));
       cout << "<<< Cadastrado >>>" << endl;
    }
    else {
       cout << "<<< \"Não\" cadastrado >>>" << endl;
    }

    cout << endl;
  }

  cout << endl;
}

void listar(vector<string> alunos) {
  cls();

  int i, n = alunos.size();
  for(i=0; i<n; i+=3) {
    cout << ((i/3)+1) << "o. aluno:" << endl;
    cout << toString(alunos[i], stod(alunos[i+1]), stod(alunos[i+2]));
  }
  pause();
}

void excluir(vector<string> *alunos) {
  cls();

  int i, n = alunos->size() / 3;

  while (true) {
    do {
      cout << "Ordem (-1 para encerrar): ";
      cin >> i;
    } while ((i < -1) || (i > n) || (n == 0));

    if (i == -1)
        break;

    cout << alunos->at((i-1)*3) << endl;
    if (confirmou("Excluir (S/*)? ")) {
       alunos->erase(alunos->begin()+((i-1)*3));
       alunos->erase(alunos->begin()+((i-1)*3));
       alunos->erase(alunos->begin()+((i-1)*3));
       cout << "<<< Excluído >>>" << endl;
    }
    else {
       cout << "<<< Mantido >>>" << endl;
    }
    cout << endl;
  }
}
